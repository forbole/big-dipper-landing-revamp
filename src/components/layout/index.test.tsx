import React from 'react';
import { render, screen } from '@testing-library/react';
import Component from '.';

jest.mock('..', () => ({
  Nav: (props: JSX.IntrinsicAttributes & React.ClassAttributes<HTMLDivElement> & React.HTMLAttributes<HTMLDivElement>) => <div data-testid="Nav" {...props} />,
  Footer: (props: JSX.IntrinsicAttributes & React.ClassAttributes<HTMLDivElement> & React.HTMLAttributes<HTMLDivElement>) => <div data-testid="Footer" {...props} />,
}));
// ==================================
// unit tests
// ==================================
describe('Layout', () => {
  it('matches snapshot', async () => {
    const { container } = render(
      <Component>
        <div>hello world</div>
      </Component>,
    );
    expect(screen.getByText(
      /hello world/i,
    )).toBeInTheDocument();
    expect(container.querySelector('main')).not.toBeNull();
    expect(screen.getByTestId('Nav')).toBeInTheDocument();
    expect(screen.getByTestId('Footer')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
