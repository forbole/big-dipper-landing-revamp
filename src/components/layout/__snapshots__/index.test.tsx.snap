// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <main
    class="emotion-0"
  >
    <div
      data-testid="Nav"
    />
    <div
      class="emotion-1"
    >
      <div>
        hello world
      </div>
    </div>
    <div
      data-testid="Footer"
    />
  </main>
</div>
`;
